src/content/docs/in-depth/customizing-prompts.mdx

```diff
@@ -8,7 +8,7 @@ sidebar:
 Implementers of the trait `SimplePrompt` take a `Prompt` as argument. A `Prompt` references a `PromptTemplate` and can render, with optional context, to something that goes to an LLM.
 
 :::warn
-Avoid creating compiled templates during a pipeline or under load. The repository is managed internally by a `RwLock` and this _will_ deadlock.
+Avoid creating **compiled** templates during a pipeline or under load. The repository is managed internally by a `RwLock` and this _will_ deadlock.
```

Comment:
Do you mean deadlock or starve here? How would the increased load cause a deadlock?

Reply:
Hmm you might be right. The repository is in a lazy static with an RW lock, so if you create precompiled templates inside a pipeline, while others are trying to read it, that will impact performance. I suspect both can happen, but starvation might be more likely.